src/stan_math_backend/Transform_Mir.ml

```diff
@@ -6,53 +6,42 @@ let use_opencl = ref false
 let opencl_triggers =
   String.Map.of_alist_exn
     [ ( "normal_id_glm_lpdf"
-      , ( [0; 1]
-        , [ (* Array of conditions under which to move to OpenCL *)
-            ([1], (* Argument 1 is data *)
-                  [(1, UnsizedType.UMatrix)])
-          (* Argument 1 is a matrix *)
-           ] ) )
-    ; ( "bernoulli_logit_glm_lpmf"
-      , ([0; 1], [([1], [(1, UnsizedType.UMatrix)])]) )
-    ; ( "categorical_logit_glm_lpmf"
-      , ([0; 1], [([1], [(1, UnsizedType.UMatrix)])]) )
-    ; ( "neg_binomial_2_log_glm_lpmf"
-      , ([0; 1], [([1], [(1, UnsizedType.UMatrix)])]) )
-    ; ( "ordered_logistic_glm_lpmf"
-      , ([0; 1], [([1], [(1, UnsizedType.UMatrix)])]) )
-    ; ("poisson_log_glm_lpmf", ([0; 1], [([1], [(1, UnsizedType.UMatrix)])]))
-    ]
+      , [ (* Array of conditions under which to move to OpenCL *)
+          [(1, UnsizedType.UMatrix)]
+        (* Argument 1 is a matrix *)
+         ] )
+    ; ("bernoulli_logit_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
+    ; ("categorical_logit_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
+    ; ("neg_binomial_2_log_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
+    ; ("ordered_logistic_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
+    ; ("poisson_log_glm_lpmf", [[(1, UnsizedType.UMatrix)]]) ]
 
 let opencl_suffix = "_opencl__"
 
 let to_matrix_cl e =
   Expr.Fixed.{e with pattern= FunApp (StanLib, "to_matrix_cl", [e])}
 
+let to_matrix_cl_non_scalar e =
+  match Expr.Typed.type_of e with
+  | UnsizedType.UInt | UReal -> e
+  | _ -> to_matrix_cl e
```

Comment:
We will only transfer non-scalar values to OpenCL. Scalars are handled by the kernel generator.

Reply:
This could be a comment in the code